datasets/acute-care/extract_trio_vcf.py

```diff
@@ -0,0 +1,171 @@
+            if family_missing:
+                logging.info(
+                    f"Family {family} is not fully represented in the data. "
+                    f"Samples missing: {sorted(family_missing)}"
+                )
+
+    # problem, none present
+    else:
+        logging.error(
+            f"No requested samples were present in the MT, please check format matches '{samples_in_mt.pop()}'"
+        )
+
+    raise NotAllSamplesPresent("please check logging messages for details")
+
+
+def get_all_unique_members(family_dict: dict) -> set:
+    """
+    pulls all unique individual members from a dict and returns as a set
+    """
+    return set(chain.from_iterable(family_dict.values()))
+
+
+def read_mt(mt_location: str, reference: str = "GRCh38") -> hail.MatrixTable:
+    assert os.path.exists(mt_location), f"Path to the mt doesn't exist: {mt_location}"
+
+    # initiate Hail expecting GRCh38
+    hl.init(default_reference=reference)
+
+    # open full MT (note, not all read in, this is done lazily with spark)
+    return hl.read_matrix_table(mt_location)
+
+
+def obtain_mt_subset(matrix: hail.MatrixTable, samples: list) -> hail.MatrixTable:
+    """
+    implements the actual subsetting of the MT
+    """
+
+    # filter the full dataset to only the samples we're interested in
+    return matrix.filter_cols(hl.literal(samples).contains(matrix["s"]))
+
+
+@click.command()
+@click.option(
+    "--json-str",
+    help="this is the formatted json string of family_sample_lookups",
+    type=click.STRING,
+)
+@click.option(
+    "--dataset", help="name of the dataset to use (gcp bucket names)", type=click.STRING
+)
+@click.option(
+    "--ref", "reference", help="name of GRChXX reference to use", default="GRCh38", type=click.STRING
+)
+@click.option(
+    "--multi_fam",
+    "multi_fam",
+    is_flag=True,
+    default=False,
+    help="use this flag if we also want a multi-family MT written to test"
+)
+def main(json_str: str, dataset: str, reference: Optional[str], multi_fam: bool):
+    """
+    This takes the family structures encoded in the JSON str and creates a number of single-family objects in Test
+
+    The option to include a multi-family structure is useful for experimenting with operations mapping MOI patterns
+    per-family within a larger dataset. This will be useful in analysing runtime/cost of analysing a single family with
+    and without extracting from a larger dataset first
+    """
+
+    # parse the families dict from the input string, e.g. '{"fam1":["sam1","sam2"]}'
+    families_dict = json.loads(json_str)
+
+    gcp_test = f"gs://cpg-{dataset}-test"
+    gcp_test_mt_outputs = os.path.join(gcp_test, "extracted_mts")
+    gcp_test_vcf_outputs = os.path.join(gcp_test, "extracted_vcfs")
+    gcp_main = f"gs://cpg-{dataset}-main"
+    gcp_mt_full = os.path.join(gcp_main, "mt", f"{dataset}.mt")
+
+    # create the output paths as empty folders
+    os.makedirs(gcp_test_mt_outputs, exist_ok=True)
```

Comment:
I don't think `os.makedirs` works with `gs://cpg-...` paths, it creates like a structure of nested directories.

```shell
python -c "import os; os.makedirs('gs://mfranklin/test/path')"

ls -lGh */**/
# gs:/:
# total 0
# drwxr-xr-x  3 michael.franklin  448689569    96B 10 Jan 13:13 mfranklin
# 
# gs:/mfranklin/:
# total 0
# drwxr-xr-x  3 michael.franklin  448689569    96B 10 Jan 13:13 test
# 
# gs:/mfranklin/test/:
# total 0
# drwxr-xr-x  2 michael.franklin  448689569    64B 10 Jan 13:13 path
# 
# gs:/mfranklin/test/path/:
```

Reply:
Agreed - I don't think it's going to behave as I expected so I'm pulling it out for now, and writing to the test root. Folder structure can be implemented later